services/rclone-http/default.nix

```diff
@@ -0,0 +1,75 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
```

Comment:
This module should set up rclone, exposing a HTTP interface. It is too much aware of the surrounding context.

 - It should not be aware of a caddy possibly fronting it
 - it should not hardcode its remote location (:azureblob:binary-cache-v1)
 - it should not configure a Nix substituter

That way, we can reuse this in other contexts (for example, having a Jenkins using a HTTP/WebDAV interface to write artifacts)